src/main/java/org/opensearch/knn/index/KNNVectorFieldMapper.java

```diff
@@ -329,8 +353,9 @@ public int getDimension() {
     protected final String spaceType;
     protected final String m;
     protected final String efConstruction;
-    private final Integer dimension;
+    protected final Integer dimension;
     protected final KNNMethodContext knnMethod;
+    protected final ModelContext modelContext;
 
     public KNNVectorFieldMapper(String simpleName, MappedFieldType mappedFieldType, MultiFields multiFields,
```

Comment:
For Another review: This class is doing so many things ( inner class ). May be we should start refactoring this. It is difficult to follow if same variable name is used in multiple inner class with different type, ex: dimension,

Reply:
Yes, I agree this class is getting a little unwieldy.

So, for dimension, I could remove it from ModelContext and keep it outside. Do you think that would be a better simplification?

Reply:
Shall we create a TODO and an issue to refactor this class?

Reply:
Sure, will add both

Reply:
#69